### PR TITLE
Test for timeseries

### LINKED ISSUE #5864: Add test for creating a `TimeSeries` from a numpy array
### Provide a general description of the issue or problem.

It should be possible to create a `GenericTimeseries` by passing a numpy `ndarray` to `sunpy.timeseries.TimeSeries`, where the first column of the `ndarray` is assumed to be the time index. This isn't covered by our test suite, but it should be. To do this a test needs to be added with the following steps:

- Create a numpy `ndarray` with the first column having datetime64 entries, and a second column of data
- Pass it to `sunpy.timeseries.TimeSeries`
- Check that the returned object has type `GenericTimeSeries`

### PR DESCRIPTION
**Pull request by akash5100**
## PR Description
Creating a `GenericTimeseries` by passing a `numpy.array` to `sunpy.timeseries.TimeSeries` 
- closes #5864 

## PR Checklist

- [x] I have followed the guidelines in the [Contributing document](https://docs.sunpy.org/en/latest/dev_guide/contents/newcomers.html)
- [x] Changes follow the coding style of this project
- [x] Changes have been formatted and linted
- [x] Changes pass pytest style unit tests (and `pytest` passes).
- [ ] Changes include any required corresponding changes to the documentation
- [ ] Changes have been rebased and squashed to the minimal number (typically 1) of relevant commits
- [ ] Changes have a descriptive commit message with a short title
- [x] I have added a `Fixes #XXXX -` or `Closes #XXXX -` comment to auto-close the issue that your PR addresses

### CODE REVIEW

File: sunpy/timeseries/timeseriesbase.py [hunk 1]
```diff
@@ -699,3 +701,28 @@ def _parse_file(cls, filepath):
             The path to the file you want to parse.
         """
         raise NoMatchError(f'Could not find any timeseries sources to parse {filepath}')
+
+
+def test_timeseries_array():
```
Comment (nabobalis) [inline sunpy/timeseries/timeseriesbase.py:706]: This should go into a test file.

Comment (akash5100) [reply sunpy/timeseries/timeseriesbase.py]: Sorry, by mistake.

File: sunpy/timeseries/tests/test_timeseriesbase.py [hunk 2]
```diff
@@ -943,6 +944,19 @@ def test_lyra_plot(lyra_test_ts):
         np.testing.assert_array_equal(ax.lines[0].get_ydata(), lyra_test_ts.to_array().T[i])
 
 
+def test_timeseries_array():
+    # Create a numpy ndarray with a first column having datetime64 entries, and a second column of data
+    # NEED HELP: instead of `datetime.date(2017, 1, 1)` should I use `datetime.datetime.today()`?
```
Comment (nabobalis) [inline sunpy/timeseries/tests/test_timeseriesbase.py:949]:
```suggestion
```

**Review by dstansby — CHANGES_REQUESTED** [review]
It looks like the tests are failing on this. Could you take a look at the test failure, and fix it? If you don't understand, feel free to leave a comment with the error message in the failing test and we can try and help.

**Comment by akash5100** [comment]
> It looks like the tests are failing on this. Could you take a look at the test failure, and fix it? If you don't understand, feel free to leave a comment with the error message in the failing test and we can try and help.

The test is failing due to the following error:
```
WARNING: SunpyUserWarning: Unknown units for 0 [sunpy.timeseries.timeseriesbase]
```
and I am not sure where I am wrong, most probably here:
> data = np.array([[date, 1], [date, 2], ])
   ts = sunpy.timeseries.TimeSeries(data)

**Comment by akash5100** [comment]
> Ah yes, you will need to add units when making the `TimeSeries`. There's an example showing how at https://docs.sunpy.org/en/stable/api/sunpy.timeseries.GenericTimeSeries.html#sunpy.timeseries.GenericTimeSeries

I tried to add units just like the example (but they are optional )
> header = {}
units = {'intensity': u.W/u.m**2} 

and passed it
> ts = sunpy.timeseries.TimeSeries(data, header, units )

and still same warning.

**Comment by nabobalis** [comment]
It might be worth debugging that by breakpointing into the class and see what is going on.

**Comment by akash5100** [comment]
I created DataFrame with the help of `np.array` and a datetime64 `np.array` whose length is equal, and everything is fine. 
In this test can we use `pd.DataFrame`?

**Comment by akash5100** [comment]
- With Dataframe
Screenshot from 2022-03-29 12-51-42
- With numpy array
Screenshot from 2022-03-29 12-51-55

The unit label on Y-axis are missing when I use the numpy array, that is the reason the tests are failing. I am searching for a way to create a Dataframe like DS using `np.array`.

**Comment by akash5100** [comment]
> That's fine, I would expect the label to not be set for a numpy array.
> 
> If we are planning to test that numpy arrays work as input, we also need to update the documentation for timeseries

I am sorry, this took time more than expected, I am fairly new to pytest and unittest. This PR helped me learn about tests and their uses and why we need them. Thank you for waiting 😄

So, should I include updating documentation in this PR?

**Comment by nabobalis** [comment]
> > That's fine, I would expect the label to not be set for a numpy array.
> > If we are planning to test that numpy arrays work as input, we also need to update the documentation for timeseries
> 
> I am sorry, this took time more than expected, I am fairly new to pytest and unittest. Do these changes look good? This PR helped me learn about tests and their uses and why we need them. Thank you for waiting 😄

I will look at it shortly.

> So, should I include updating documentation in this PR?

Sure.

**Comment by akash5100** [comment]
> Sure.

For the documentation of the time series, Can you tell me which part needs changes? and how?

**Comment by nabobalis** [comment]
> For the documentation of the time series, Can you tell me which part needs changes? and how?

You need to change https://docs.sunpy.org/en/stable/api/sunpy.timeseries.GenericTimeSeries.html#sunpy.timeseries.GenericTimeSeries to say for the data variable, pd.dataframe or numpy array.

**Review by dstansby — APPROVED** [review]
Thanks - this hits the line that needed covering in the tests 🎉

**Comment by akash5100** [comment]
Thank you @dstansby, for the review

> You need to change https://docs.sunpy.org/en/stable/api/sunpy.timeseries.GenericTimeSeries.html#sunpy.timeseries.GenericTimeSeries to say for the data variable, pd.dataframe or numpy array.

Should, I add this in a new PR?

**Comment by nabobalis** [comment]
> Thank you @dstansby, for the review
> 
> > You need to change [docs.sunpy.org/en/stable/api/sunpy.timeseries.GenericTimeSeries.html#sunpy.timeseries.GenericTimeSeries](https://docs.sunpy.org/en/stable/api/sunpy.timeseries.GenericTimeSeries.html#sunpy.timeseries.GenericTimeSeries) to say for the data variable, pd.dataframe or numpy array.
> 
> Should, I add this in a new PR?

Yes please.

**Pull request merged**